### PR TITLE
Use cached stats result to lookup names

### PR DESCRIPTION
**Pull request by tschroeder-zendesk**
Code was doing a client lookup for each name making it very slow to determine which node to use.

### CODE REVIEW

File: curator/actions.py [hunk 1]
```diff
@@ -1901,7 +1901,7 @@ def most_available_node(self):
         mvn_id = None
         nodes = self.client.nodes.stats()['nodes']
         for node_id in nodes:
-            name = self.client.nodes.stats()['nodes'][node_id]['name']
+            name = nodes[node_id]['name']
```
Comment (untergeek) [inline curator/actions.py]: 🤦‍♂️ Thanks for this.  Can't believe I didn't think of that.

Comment (untergeek) [reply curator/actions.py]: Please add a comment to the Changelog on this in the bug fixes section.

Comment (tschroeder-zendesk) [reply curator/actions.py]: Done!

File: Changelog.rst [hunk 2]
```diff
@@ -8,34 +8,35 @@ Changelog
 **Bug Fixes**
 
   * The ``restore_check`` function did not work properly with wildcard index
-    patterns.  This has been rectified, and an integration test added to 
+    patterns.  This has been rectified, and an integration test added to
     satisfy this.  Reported in #989 (untergeek)
+  * Re-use cached response for nodes in most_available_node function
```
Comment (untergeek) [inline Changelog.rst]: Can you mention the PR # and your github handle in parenthesis, like the other lines?

File: curator/actions.py (hunk 1 above)
Comment (untergeek) [reply curator/actions.py]: I think know why it's like that now.  I probably cut and pasted what I was using in `ipython` to test (it wasn't in a loop, there).

File: Changelog.rst (hunk 2 above)
Comment (tschroeder-zendesk) [reply Changelog.rst]: 👍

**Pull request merged**